### PR TITLE
Expose FP8 API to Python

### PR DESCRIPTION
**Pull request by ahsan-ca**
This exposes autocast_fp8_pass to Python.

### CODE REVIEW

File: src/py/migraphx_py.cpp [hunk 1]
```diff
@@ -581,6 +582,15 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
           py::arg("t"),
           py::arg("calibration") = std::vector<migraphx::parameter_map>{},
           py::arg("ins_names")   = std::unordered_set<std::string>{"dot", "convolution"});
+    m.def(
+        "autocast_fp8_pass",
+        [](migraphx::module& input_module) {
```
Comment (pfultz2) [inline src/py/migraphx_py.cpp:587]: This should be applied to the `migraphx::program` not just the module.

File: src/py/migraphx_py.cpp [hunk 2]
```diff
@@ -581,6 +582,15 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
           py::arg("t"),
           py::arg("calibration") = std::vector<migraphx::parameter_map>{},
           py::arg("ins_names")   = std::unordered_set<std::string>{"dot", "convolution"});
+    m.def(
+        "autocast_fp8_pass",
```
Comment (pfultz2) [inline src/py/migraphx_py.cpp:586]: This should probably just be named `autocast_fp8`.

File: src/py/migraphx_py.cpp [hunk 3]
```diff
@@ -581,6 +582,15 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
...
           py::arg("calibration") = std::vector<migraphx::parameter_map>{},
           py::arg("ins_names")   = std::unordered_set<std::string>{"dot", "convolution"});
+    m.def(
+        "autocast_fp8",
+        [](migraphx::program& p) {
+            migraphx::autocast_fp8_pass pass;
+            migraphx::module& input_module = *p.get_main_module();
+            pass.apply(input_module);
+        },
+        "Create and apply autocast_fp8_pass to a module",
```
Comment (umangyadav) [inline src/py/migraphx_py.cpp:592]:
```suggestion
        "Auto-convert FP8 parameters and return values  to Float for MIGraphX Program",
```

File: src/py/migraphx_py.cpp [hunk 4]
```diff
@@ -581,6 +582,15 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
...
+        py::arg("m"));
```
Comment (umangyadav) [inline src/py/migraphx_py.cpp:593]:
```suggestion
        py::arg("p"));
```

File: src/py/migraphx_py.cpp [hunk 5]
```diff
@@ -581,6 +582,15 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
           py::arg("t"),
           py::arg("calibration") = std::vector<migraphx::parameter_map>{},
           py::arg("ins_names")   = std::unordered_set<std::string>{"dot", "convolution"});
+    m.def(
+        "autocast_fp8",
+        [](migraphx::program& p) {
+            migraphx::autocast_fp8_pass pass;
+            migraphx::module& input_module = *p.get_main_module();
+            pass.apply(input_module);
```
Comment (pfultz2) [inline src/py/migraphx_py.cpp:590]: Use `run_passes` to run the pass: `run_passes(*p.get_main_module(), autocast_fp8_pass{})`.

Comment (ahsan-ca) [reply src/py/migraphx_py.cpp]: Done.

File: src/py/migraphx_py.cpp (hunk 4 above)
Comment (ahsan-ca) [reply src/py/migraphx_py.cpp]: Done.

File: src/py/migraphx_py.cpp (hunk 3 above)
Comment (ahsan-ca) [reply src/py/migraphx_py.cpp]: Done.

**Review by umangyadav — APPROVED** [review]

**Pull request merged**